lib/moonshot/artifact_repository/s3_bucket_via_github_releases.rb

```diff
@@ -115,22 +116,22 @@ def upload_to_s3(file, key)
     # @raise [RuntimeError] If the file fails to download correctly after 3
     #                       attempts.
     def download_from_github(version)
+      filename = "*#{version}*.tar.gz"
       attempts = 0
-      begin
+
+      Retriable.retriable on: RuntimeError do
         # Make sure the directory is empty before downloading the release.
         FileUtils.rm(Dir.glob('*'))
 
         # Download the release and find the actual build file.
         sh_out("hub release download #{version}")
-        file = Dir.glob("*#{version}*.tar.gz").fetch(0)
+        raise "File '#{filename}' not found." unless File.file?(filename)
```

Comment:
Given that this passed travis, then I think there are probably some missing unit tests as well.